src/settings/accounts-settings.tsx

```diff
@@ -0,0 +1,346 @@
+import { SHELL_APP_ID } from '../constants';
+import { Mods, IdentityProps, CreateIdentityProps } from '../../types';
+import { useAccountStore } from '../store/account/store';
+import AccountsList from './components/account-settings/accounts-list';
+import PrimaryAccountSettings from './components/account-settings/primary-account-settings';
+import SettingsSentMessages from './components/account-settings/settings-sent-messages';
+import PasswordRecoverySettings from './components/account-settings/password-recovery-settings';
+import Delegates from './components/account-settings/delegates';
+import PersonaSettings from './components/account-settings/persona-settings';
+import UsePersona from './components/account-settings/use-persona';
+// external accounts not yet activated, graphical part is complete
+// import ExternalAccount from './components/account-settings/external-account-settings';
+// import AdvancedSettings from './components/account-settings/advanced-settings';
+// import DownloadMessages from './components/account-settings/download-messages';
+
+export const DisplayerHeader: FC<{
+	onSave: (mods: Record<string, unknown>) => void;
+	mods: Record<string, unknown>;
+	t: TFunction;
+}> = ({ onSave, mods, t }) => {
+	const accountLabel: string = useMemo(() => t('label.accounts', 'Accounts'), [t]);
+	const buttonLabel = useMemo(() => t('label.save', 'Save'), [t]);
+	return (
+		<Container
+			orientation="vertical"
+			mainAlignment="flex-start"
+			height="fit"
+			background="gray5"
+			padding={{ bottom: 'medium', right: 'medium' }}
+		>
+			<Row orientation="horizontal" width="100%">
+				<Row
+					padding={{ all: 'small' }}
+					mainAlignment="flex-start"
+					width="50%"
+					crossAlignment="flex-start"
+				>
+					<Text size="large" weight="regular">
+						{accountLabel}
+					</Text>
+				</Row>
+				<Row
+					padding={{ all: 'small' }}
+					width="50%"
+					mainAlignment="flex-end"
+					crossAlignment="flex-end"
+				>
+					<Button
+						label={buttonLabel}
+						color="primary"
+						onClick={onSave}
+						disabled={Object.keys(mods).length === 0}
+					/>
+				</Row>
+			</Row>
+			<Divider />
+		</Container>
+	);
+};
+
+const AccountsSettings = (): ReactElement => {
+	const [mods, setMods] = useState<Mods>({});
+	const [t] = useTranslation();
+	const accountSettings = useUserAccount();
+	const [activeDelegateView, setActiveDelegateView] = useState('0');
+	const addMod = useCallback(
+		(
+			type: 'props' | 'prefs' | 'identity',
+			id: string | undefined,
+			key: any,
+			value?: string,
+			action?: string,
+			deleteList?: string[],
+			createList?: CreateIdentityProps
+		) => {
+			setMods((m) => ({
+				...m,
+
+				[type]: { id, action, deleteList, createList, ...m?.[type], [key]: value }
+			}));
+		},
+		[]
+	);
+
+	const createIdentities = useCallback(
+		(createList) => {
+			addMod('identity', undefined, undefined, undefined, 'create', undefined, createList);
```

Comment:
I always prefer to pass object as argument instead of passing props like this with values like undefined or blank.By object it will be lot better to handle

Reply:
the object argument makes sense